system-x/services/ldap/src/main/java/software/tnb/ldap/validation/LdapValidation.java

```diff
@@ -0,0 +1,64 @@
+package software.tnb.ldap.validation;
+
+import software.tnb.common.validation.Validation;
+import software.tnb.ldap.account.LdapAccount;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.unboundid.ldap.sdk.LDAPConnection;
+import com.unboundid.ldap.sdk.LDAPException;
+import com.unboundid.ldap.sdk.SearchRequest;
+import com.unboundid.ldap.sdk.SearchResult;
+import com.unboundid.ldap.sdk.SearchResultEntry;
+import com.unboundid.ldap.sdk.SearchScope;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LdapValidation implements Validation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(LdapValidation.class);
+    private LdapAccount ldapAccount;
+    private LDAPConnection ldapConnection;
+    private String ldapUrl;
+
+    public LdapValidation(LDAPConnection ldapConnection, LdapAccount ldapAccount, String ldapUrl) {
+        this.ldapConnection = ldapConnection;
+        this.ldapAccount = ldapAccount;
+        this.ldapUrl = ldapUrl;
+    }
+
+    public List<String> getEntries(String objectClass) {
+        List<String> entries = new ArrayList<>();
+        try {
+            ldapConnection.connect(ldapUrl.substring((ldapUrl.lastIndexOf('/') + 1), ldapUrl.lastIndexOf(':')),
+                Integer.parseInt(ldapUrl.substring(ldapUrl.lastIndexOf(':') + 1)));
+            ldapConnection.bind(ldapAccount.username(), ldapAccount.password());
+            String searchBaseDN = "dc=example,dc=org";
+            String searchFilter = "(objectClass=" + objectClass + ")";
+            SearchRequest searchRequest = new SearchRequest(searchBaseDN, SearchScope.SUB, searchFilter);
+            SearchResult searchResult = ldapConnection.search(searchRequest);
+            for (SearchResultEntry entry : searchResult.getSearchEntries()) {
+                entries.add(entry.getDN());
+            }
+            ldapConnection.close();
+        } catch (LDAPException e) {
+            LOG.error("Error when connecting to ldap server");
+            throw new RuntimeException("Error when connecting to ldap server");
+        }
+        return entries;
+    }
+
+    public String getLdapUsername() {
+        return ldapAccount.username();
+    }
+
+    public String getLdapPassword() {
+        return ldapAccount.password();
+    }
```

Comment:
these two can be accessed in the test through `ldap.account()`, so it is not needed to define them here